features/storage/kvstore/securestore/SecureStore.cpp

```diff
@@ -776,21 +797,36 @@ int SecureStore::init()
 int SecureStore::deinit()
 {
     _mutex.lock();
+    int ret;
     if (_is_initialized) {
-        mbedtls_entropy_free(static_cast<mbedtls_entropy_context *>(_entropy));
-        delete static_cast<mbedtls_entropy_context *>(_entropy);
-        delete static_cast<inc_set_handle_t *>(_inc_set_handle);
-        delete _scratch_buf;
-        // TODO: Deinit member KVs?
+        if (_entropy) {
+            mbedtls_entropy_free(static_cast<mbedtls_entropy_context *>(_entropy));
+            delete static_cast<mbedtls_entropy_context *>(_entropy);
```

Comment:
Feel free to get rid of these horrible casts any time. I can't see any need for `void *` here, just need some forward declarations in the header.